### PR TITLE
feat: supporting changing module preload mode

### LINKED ISSUE #2518: Add support for creating module preloads for only the static dependencies
It currently seems to also create `modulepreload`'s for the dynamic imports. I'd like these to only be loaded as needed.

### PR DESCRIPTION
**Pull request by Bubblyworld**
Fixes #2518. There's an `esm.sh` provider test that's failing currently until
we pull in this downstream fix in the generator:
https://github.com/jspm/generator/pull/295

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -361,16 +361,37 @@ function getCacheMode(flags: Flags): "offline" | boolean {
...
     );
 
   if (flags.cache === "offline") return "offline";
   if (flags.cache === "online") return true;
   return false;
 }
 
+const validPreloadModes = ["static", "dynamic", "no-preloads"];
+function getPreloadMode(flags: Flags): boolean | string {
+  if (!flags.preload) return "static";
```
Comment (guybedford) [inline src/utils.ts]: When no flag is set, we shouldn't do any preloading at all.

File: src/utils.ts [hunk 2]
```diff
@@ -361,16 +361,37 @@ function getCacheMode(flags: Flags): "offline" | boolean {
...
+  if (!validPreloadModes.includes(flags.preload))
```
Comment (guybedford) [inline src/utils.ts:381]: Can we allow a "boolean" form ie `--preload` without a value? Perhaps we need to preparse this in the CLI handling?

File: src/utils.ts (hunk 1 above)
Comment (Bubblyworld) [reply src/utils.ts]: Mmm, think I misunderstood what you meant by defaulting to static. You mean default to static only if the flag is present at all?

Comment (guybedford) [reply src/utils.ts]: Yes exactly, preloading shouldn't be blanket enabled by default.

**Pull request merged**